### ISSUE TITLE
"Transfer learning" from previous dlib dnn

### ISSUE THREAD

**Issue by Galisela** [davisking/dlib#1143]
Hello,

I have trained an object detector easily using dlib's vehicle detection example. I now have a larger labeled dataset and wanted to finetune the model.

I deleted the initialization of the mmod layer deserialized  the previous detector and ran normally.

When doing this I get errors about overlapping truth rectangles which I assume are related to the mmod_options, that were set according to the smaller data set,  specifically  overlaps_nms.

My questions are:
1. Am I right to assume my new dataset just has more overlap than the old and that is the issue?
2. Does the warning refer to the truth rectangles in the labeling or to the detected rectangles?
3. Can I access the mmod_options of the network without re initializing the weights or should I just retrain the entire network? 

Thanks a lot in advance!

This is the warning I get:
"Warning, ignoring object.  We encountered a truth rectangle located at [(163, 164) (201, 197)] that is suppressed by non-max-suppression because it is overlapped by another truth rectangle located at [(120, 119) (189, 191)] (IoU:0.116703, Percent covered:0.570136)."

**Comment by davisking** [comment]
Those warnings are telling you that the non-max suppression setting you have configured in the mmod_options is so aggressive that it's impossible for the detector you are trying to train to detect all the objects in your training dataset.  You could just ignore it, maybe it doesn't matter to you.  Or maybe it matters, I can't say.  That's why the warning is output.

Yes, you can access any layer and mutate it however you like.  Look at the documentation for the add_loss_layer object.  All the member functions are documented.

**Comment by Galisela** [comment]
I was mistakenly looking for it on loss.h thanks for referring me to the correct function!

**Comment by faboules** [comment]
I was mistakenly look for it on loss.h thanks for reffering me to the correct function !
